auto-milestone/main.go

```diff
@@ -0,0 +1,230 @@
+	pflag.BoolVar(&doPreview, "preview", false, "Only determine the milestone but don't set it")
+	pflag.BoolVar(&listInputs, "list-inputs", false, "Show a list of all available inputs")
+	pflag.Parse()
+
+	logger.Info().Msgf("Operating inside %s", repo)
+
+	rawPRNumber := pflag.Arg(0)
+	if rawPRNumber == "" {
+		logger.Fatal().Msg("No PR specified")
+		return
+	}
+	if parsed, err := strconv.ParseInt(rawPRNumber, 10, 32); err != nil {
+		logger.Fatal().Err(err).Msg("Failed to parse PR number")
+		return
+	} else {
+		prNumber = int(parsed)
+	}
+
+	// Determine the base-branch of that pull request
+	tk, err := toolkit.Init(
+		ctx,
+	)
+	if err != nil {
+		logger.Fatal().Err(err).Msg("Failed to initialize toolkit")
+	}
+
+	if listInputs {
+		tk.ShowInputList()
+		return
+	}
+	defer func() {
+		if err := tk.SubmitUsageMetrics(ctx); err != nil {
+			logger.Warn().Err(err).Msg("Failed to submit usage metrics")
+		}
+	}()
+
+	elems := strings.Split(repo, "/")
+	repoOwner := elems[0]
+	repoName := elems[1]
+
+	gh := tk.GitHubClient()
+
+	pr, _, err := gh.PullRequests.Get(ctx, repoOwner, repoName, prNumber)
+	if err != nil {
+		logger.Fatal().Err(err).Msg("Failed to fetch pull-request")
+	}
+	prTargetBranch := pr.GetBase()
+	if prTargetBranch == nil {
+		logger.Fatal().Msg("The requested pull-request has no target branch")
+		return
+	}
+
+	var targetMilestoneName string
+
+	content, _, _, err := gh.Repositories.GetContents(ctx, prTargetBranch.GetRepo().GetOwner().GetLogin(), prTargetBranch.GetRepo().GetName(), "package.json", &github.RepositoryContentGetOptions{Ref: prTargetBranch.GetRef()})
+	if err != nil {
+		logger.Fatal().Err(err).Msg("Failed to retrieve package.json of PR base")
+		return
+	}
+	if cnt, err := content.GetContent(); err != nil {
+		logger.Fatal().Err(err).Msg("Failed to retrieve package.json of PR base")
+		return
+	} else {
+		if v, err := versionFromPackage(cnt); err != nil {
+			logger.Fatal().Err(err).Msg("Failed to determine base version")
+			return
+		} else {
+			targetMilestoneName = v
+		}
+	}
+
+	prMilestone := pr.GetMilestone()
+
+	if prMilestone != nil {
+		logger.Info().Msgf("Current milestone: %s", prMilestone.GetTitle())
+	}
+	logger.Info().Msgf("Target milestone name: %s", targetMilestoneName)
+	milestone, err := tk.GitHubGQLClient().GetMilestoneByTitle(ctx, repoOwner, repoName, targetMilestoneName)
+	if err != nil {
+		logger.Fatal().Msgf("Failed to find milestone matching `%s`", targetMilestoneName)
+	}
+
+	a := determineAction(ctx, pr, prMilestone, milestone)
+
+	if doPreview {
+		logger.Info().Msgf("The following action would be performed: %v", a)
+		return
+	}
+
+	switch a.Type {
+	case actionTypeNoop:
+		logger.Info().Msgf("No action necessary.")
+		return
+	case actionTypeSetToMilestone:
+		logger.Info().Msgf("Updating PR to %s", a.Milestone)
+		if a.Milestone != nil {
+			if _, _, err := gh.Issues.Edit(ctx, repoOwner, repoName, prNumber, &github.IssueRequest{
```

Comment:
When playing with `gh` client I always wonder if err is enough or if we should also handle the status number as well 😅

Reply:
Will change that and run some basic smoke-tests that `< 300` as OK-condition is ok enough ;)